### DISCUSSION TITLE
How to implement billboard effect by using filament

### DISCUSSION THREAD

**Discussion by dandingol03** [google/filament#6784] [Q&A] [answered]
**Description**
Hi, i have looked over several issues about implement `billboard`. I come here to get some help because none of them work. I have been stucked for a long time.
**Context**
There is a text mesh in the glb model, and user can rotate the model in any axis they want. The key point is to keep the text mesh face to the mobile screen always. An idea is to know the world position of the mesh center. There are two methods to implement `billboard` after that:

1. Render the text mesh manually by emitting four other vertex in the geometry shader. Theses four vertex compose the text mesh which faces to the mobile screen always. However the geometry shader can't be support in filament
2. Modify the text mesh vertecies position in vertex shader. Get the world position of each one in vertex shader, and modify its position by the distance from this vertex to the center of text mesh and camera forward direction. The difficult part is how can i get the world positiion of the center of text mesh, because filament can only get `WorldTransform` by using transform manager.

**Comment by bejado** [comment, marked as answer]
I'm not sure you even need to modify shaders for this. You could adjust the rotation of the text mesh entity using the transform manager so it always points towards the camera.

**Reply by dandingol03** [reply]
yes, i render the mesh as a quad which is composed of four vertex and i compute these four vertex coordinate based the mesh center position every frame to make the mesh normal consistent with the camera forward vector
